rasa/server.py

```diff
@@ -684,9 +693,10 @@ async def execute_action(request: Request, conversation_id: Text):
 
         try:
             async with app.agent.lock_store.lock(conversation_id):
-                tracker = await get_tracker_with_session_start(
-                    app.agent.create_processor(), conversation_id
+                tracker = await app.agent.create_processor().fetch_tracker_and_update_session(
```

Comment:
out of scope: Your change really makes it obvious that we don't really care about the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) 😆 In my had head all tracker store interaction should happen in `Agent`. The processor should only work on `tracker` events and the Agent would store the results in the tracker store once the `processor` is done.

Reply:
I agree that this is odd, but as you said this is a larger refactoring